### PR TITLE
Better DNS persistence solution for systemd

### PR DESCRIPTION
**Pull request by kkovacs**
Updated documentation with a non-oneshot solution to configure systemd's resolvectl on the host machine to be able to resolve LXD containers by name.

### CODE REVIEW

**Comment by tomponline** [comment]
Please may you open this against the main branch first for the current LXD series rather than stable-5.0?

**Comment by simondeziel** [comment]
@kkovacs now that you want your changes to be incorporated into the main branch, you need to update the `base branch` setting accordingly.

I tried doing it for you but then it tells me there are no new commit as if you had not included your new contribution in what you just pushed.

File: doc/howto/network_bridge_resolved.md [hunk 1]
```diff
@@ -64,6 +64,55 @@ You must repeat the commands after each reboot and after LXD is restarted, or ma
 
 ## Make the `resolved` configuration persistent
 
+### Solution "A"
+
+You can automate the `systemd-resolved` DNS configuration, so that it is applied on system start and takes effect when LXD creates the network interface.
+
+To do so, create a `systemd` network file named `/etc/systemd/network/<network_bridge>.network` with the following content:
```
[the remark this answers is not in the document]
Comment (kkovacs) [reply doc/howto/network_bridge_resolved.md]: I tested it on a fresh install, and you are right in the sense that it doesn't get automatically noticed (but works reliably after a reboot).

But of course that's an unelegant solution. Apparently doing `networkctl reload` puts `lxdbr0` in a managed (`configured`) state, and also the `systemd-resolved` config also *looks* correct after that, but the routing to the `10.x.x..1` IP is not working yet at that point, so it needs a bit of exploration what unholy parts of systemd need to be kicked for it to properly work.

I'll amend this when I'm re-opening the PR against the right branch, as @tomponline requested.

**Comment by kkovacs** [comment]
> @kkovacs now that you want your changes to be incorporated into the main branch, you need to update the `base branch` setting accordingly.
> 
> I tried doing it for you but then it tells me there are no new commit as if you had not included your new contribution in what you just pushed.

Thanks for that, yes, something is strange, github said my branch is ~6000 commits ahead :) I'll recreate the whole branch+PR after I figure out the answer to the "what needs to be restarted" question.

**Pull request closed without merging**